ratpack-dropwizard-metrics/src/main/java/ratpack/dropwizard/metrics/internal/HttpClientMetrics.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2013 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ratpack.dropwizard.metrics.internal;
+
+import com.codahale.metrics.Gauge;
+import com.codahale.metrics.MetricRegistry;
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import ratpack.dropwizard.metrics.DropwizardMetricsConfig;
+import ratpack.dropwizard.metrics.HttpClientConfig;
+import ratpack.exec.ExecController;
+import ratpack.http.client.HttpClient;
+import ratpack.http.client.HttpClientStats;
+import ratpack.service.Service;
+import ratpack.service.StartEvent;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+
+@Singleton
+public class HttpClientMetrics implements Service, Runnable {
+
+  private static final String METRIC_PREFIX = "httpclient.";
+  private static final String TOTAL_ACTIVE_CONNECTIONS = getMetricName("total.active.connections");
+  private static final String TOTAL_IDLE_CONNECTIONS = getMetricName("total.idle.connections");
+  private static final String TOTAL_CONNECTIONS = getMetricName("total.connections");
+
+  private final HttpClient httpClient;
+  private final MetricRegistry metricRegistry;
+  private final HttpClientConfig config;
+  private final Map<String, HttpMetricGauge> gauges;
+
+  @Inject
+  public HttpClientMetrics(
+    HttpClient httpClient,
+    MetricRegistry metricRegistry,
+    DropwizardMetricsConfig config
+  ) {
+    this.httpClient = httpClient;
+    this.metricRegistry = metricRegistry;
+    this.config = config.getHttpClient();
+    this.gauges = new HashMap<>();
+  }
+
+  @Override
+  public void onStart(StartEvent event) throws Exception {
+    if (config.isEnabled()) {
+      ExecController execController = event.getRegistry().get(ExecController.class);
+      execController.getExecutor().scheduleAtFixedRate(this, 0, config.getPollingFrequencyInSeconds(), TimeUnit.SECONDS);
```

Comment:
I don't particular like this polling method of checking the HttpClient stats, but no other ideas forth coming.  It does seem to work, but open to other options.